lib/engine/game/g_18_oe/step/buy_sell_par_shares.rb

```diff
@@ -1,38 +1,102 @@
 # frozen_string_literal: true
 
 require_relative '../../../step/buy_sell_par_shares'
+require_relative '../../../action/convert'
 
 module Engine
   module Game
     module G18OE
       module Step
         class BuySellParShares < Engine::Step::BuySellParShares
           def actions(entity)
+            return corporation_actions(entity) if entity.corporation?
```

Comment:
Do you want to do all corporations or ones owned by the current entity?

Reply:
You can float major if you have 50% shares but not ownership, so need to check all corps